benchmarking/src/run.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022-2023 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+//! Library for running and reporting benchmark results.
+
+use std::{collections::HashMap, ffi::OsStr, path::PathBuf, time::Duration};
+
+use crate::{
+    result::RunResult,
+    time_bench::{Time, REPO_ROOT_PATH},
+};
+
+use tabled::settings::{
+    object::{Columns, Object, Rows},
+    width::MinWidth,
+    Alignment, Color, Modify, Style, Width,
+};
+use walkdir::WalkDir;
+
+/// A benchmark configuration and its resulting measurement.
+#[derive(Debug, Clone)]
+pub struct BenchmarkResult {
```

Comment:
Can we call this something else? Maybe just `Benchmark`? There are already two other things called "Result" that are defined in this crate, beyond the usage in `std`.

Reply:
`Benchmark` sounds like a configuration for a benchmark rather than its results, but I share your concern. How about `Measurement` or `BenchmarkMeasurement`?

Reply:
I've renamed a few things to use Measurement instead of Result, let me know if you have another suggestion.